oak-lucene/src/test/java/org/apache/jackrabbit/oak/composite/blueGreen/IndexUtils.java

```diff
@@ -142,7 +142,7 @@ public static void checkLibsIsReadOnly(Persistence p) throws RepositoryException
         try {
             libsNode.addNode("illegal");
             Assert.fail();
-        } catch (UnsupportedOperationException e) {
+        } catch (RepositoryException e) {
```

Comment:
Catch `ConstraintViolationException` instead?

Reply:
Won't change it here. This test is not responsible to test the implementation of the JCR, so it should just rely on the API contract.

Reply:
> This test is not responsible to test the implementation of the JCR, so it should just rely on the API contract.

Well, currently the test does check an implementation detail. It catches a UnsupportedOperationException. But you are right the method does not `UnsupportedOperationException` nor `ConstraintViolationException`.